### PR TITLE
Update the tp examples to make it more clear

### PR DESCRIPTION
**Pull request by guihao-liang**
# Summary

1. 添加 row parallelism 的example。导言里面介绍了，但是实战没有
2. 更新了column parallelism的example。

之前的版本不是特别清晰, 例如

```python
    for i in range(num_gpus):
        start_idx = i * chunk_size
        end_idx = start_idx + chunk_size
        # ==========================================
        # TODO 1: 沿列方向 (dim=1) 对 A 进行切片
        # ==========================================
        # weight_chunk = ???
        # gpu_weights.append(weight_chunk)
        weight_chunk = torch.zeros(in_features, chunk_size)  # 占位初始化
        gpu_weights.append(weight_chunk)
```

1. 到底要用start idx 还是直接用 chunk？
2. 初始化占位符要不要删除？

新的版本:

```python
    gpu_weights = []
    for i in range(num_gpus):
        start_idx = i * chunk_size
        end_idx = start_idx + chunk_size
        # ==========================================
        # TODO 1: 沿列方向 (dim=1) 对 A 进行切片
        # 可以使用 start_idx，end_idx 对 A 切片
        # 也可以不使用上述 idx，直接用 chunk 切片
        # torch.chunk(...)[i]
        # ==========================================
        # weight_chunk = ???
        # gpu_weights.append(weight_chunk)
        pass
```

# Test

本地 e2e 测试过了

```python
def tensor_parallel_row_sim(X: torch.Tensor, A: torch.Tensor, num_gpus: int = 2):
    """
    模拟 Column Parallel Linear: Y = X @ A
    将权重 A 沿列 (输出特征维度) 切分，分布到不同的 GPU 上计算，最后拼接。
    
    参数:
    X: 形状 (batch, in_features)
    A: 形状 (in_features, out_features)
    """
    in_features, out_features = A.shape
    assert in_features % num_gpus == 0, "input feature must be slipt"
    
    # 1. 模拟将权重加载到不同 GPU 的显存中
    # gpu_weights 是一个列表，代表各 GPU 本地保存的权重分片
    x_splits = torch.chunk(X, num_gpus, dim=-1)
    gpu_weights = torch.chunk(A, num_gpus, dim=0)
        
    # 2. 模拟各 GPU 并行前向计算
    # X 进行 column split
    # A 进行 row split
    gpu_outputs = []
    for i in range(num_gpus):
        # ==========================================
        # TODO 2: 每张卡使用自己本地的权重分片，对输入 X 进行矩阵乘法计算
        # ==========================================
        local_out = x_splits[i] @ gpu_weights[i]  # 占位初始化
        gpu_outputs.append(local_out)
        
    # 3. 模拟 All-Reduce 通信操作
    # ==========================================
    # TODO 3: sum each prod result
    # ==========================================
    Y_gathered = sum(gpu_outputs)
    return Y_gathered
```

### CODE REVIEW

**Review by lynnyulinlin-debug — CHANGES_REQUESTED** [review]
感谢你的关注和贡献！整体方向非常正确，Row Parallel 的补充刚好补齐了 TP 两种切分方式的讲解，避免了"讲了理论但没有落地练习"的断层感。

我这边有几条建议，部分属于当前 PR 的收尾要求，部分可以后续独立处理：

建议在当前 PR 中处理（建议合入前完成）：

    修正 tensor_parallel_row_sim 的 docstring：当前仍误写为 "Column Parallel Linear"，需改为 "Row Parallel Linear"。

    同步更新 docs/ 目录：确保文档与代码一致。

    运行项目提供的验证脚本：确保结果符合仓库约定。

可选优化（可后续 PR 跟进）：

    sum(y_chunks) 的写法：功能上正确，但 Python 内置 sum() 作用于张量列表时不够直观，建议改为更明确的 torch.stack(y_chunks, dim=0).sum(dim=0) 或显式循环累加。

    变量命名：a_local = x_chunks[i] @ a_chunks[i] 中的 a_local 实际存储的是计算结果而非权重，命名可以更清晰。

    术语统一：X / A 的切分注释可以进一步统一，例如统一使用"沿特征维度切分"或"沿输入维度切分"。

关于 TODO 统一接口的说明：

代码中 TODO 部分的接口设计（输入输出格式、函数签名风格等），我已在后续章节规划中有统一考虑，这次 PR 可以先保持现有结构，不必一次性全部收口。

如果希望加快合入速度，建议优先完成"当前 PR 中处理"的三项，其他优化点后续单独跟进即可。再次感谢！🙌
